api/src/main/java/jakarta/data/metamodel/constraint/GreaterThanOrEqualRecord.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel.constraint;
+
+import jakarta.data.metamodel.restrict.Operator;
+
+import java.util.Objects;
+
+record GreaterThanOrEqualRecord<T extends Comparable<T>>(T bound)
+        implements GreaterThanOrEqual<T> {
+    public GreaterThanOrEqualRecord {
+        Objects.requireNonNull(bound, "Lower bound must not be null");
+    }
+
+    @Override
+    public Operator operator() {
+        return Operator.GREATER_THAN_EQUAL;
+    }
+
+    @Override
+    public String toString() {
+        return bound.toString();
```

Comment:
Distinguish GreaterThanEqual.toString output from EqualTo.toString output
```suggestion
        return ">= " + bound.toString();
```